cartography/intel/gcp/storage.py

```diff
@@ -0,0 +1,223 @@
+import logging
+from collections import namedtuple
+
+import googleapiclient.discovery
+from oauth2client.client import ApplicationDefaultCredentialsError
+from oauth2client.client import GoogleCredentials
+
+from cartography.intel.gcp import compute
+logger = logging.getLogger(__name__)
+
+
+def get_gcp_bucket_metadata(storage, bucket):
+    """
+    Retrieves metadata about the given bucket
+
+    :type storage: A storage resource object
+    :param storage: The storage resource object created by googleapiclient.discovery.build()
+
+    :type bucket: str
+    :param bucket: Google Cloud Bucket name
+
+    :rtype: dict
+    :return: Metadata for specified bucket
+    """
+    try:
+        req = storage.buckets().get(bucket=bucket)
+        res = req.execute()
+        return res
+    except HttpError as e:
+        reason = compute._get_error_reason(e)
+        if reason == 'notFound':
+            logger.debug(
+                ("The bucket %s was not found - returned a 404 not found error."
+                 "Full details: %s"), bucket, e, )
+            return None
+        elif reason == 'forbidden':
+            logger.debug(
+                ("You do not have storage.bucket.get access to the bucket %s. "
+                 "Full details: %s"), bucket, e, )
+            return None
+        else:
+            raise
+
+
+def get_gcp_buckets(storage, project):
+    """
+    Returns a list of storage objects within some given project
+
+    :type storage: A storage resource object
+    :param storage: The storage resource object created by googleapiclient.discovery.build()
+
+    :type project: str
+    :param project: The Google Project name that you are retrieving buckets from
+    :rtype: list
+    :return: List of storage objects
+    """
+    try:
+        req = storage.buckets().list(project=project)
+        res = req.execute()
+        all_objects = []
+        all_objects.extend(res.get('items', []))
+        return all_objects
```

Comment:
As I mentioned in my comment, let's try return `res` itself for now so that we have access to `res.id` in the `transform_*` function to get the `project_id`.

Reply:
Hmm.. Forgive me if I'm missing it but there doesn't seem to be any `project_id` that is able to be parsed from the storage resource object. I got the following object:
```json 
{
 "kind": "storage#buckets",
 "items": [
  {
   "kind": "storage#bucket",
   "id": "bucket_name",
   "selfLink": "https://www.googleapis.com/storage/v1/b/bucket_name",
   "projectNumber": "some_num",
   "name": "bucket_name",
   "timeCreated": "some_time",
   "updated": "some_time",
   "metageneration": "1",
   "iamConfiguration": {
    "bucketPolicyOnly": {
     "enabled": false
    },
    "uniformBucketLevelAccess": {
     "enabled": false
    }
   },
   "location": "US",
   "locationType": "multi-region",
   "defaultEventBasedHold": false,
   "storageClass": "STANDARD",
   "etag": "CAE="
  }
 ]
}
```
Am I missing something? Maybe I'm not looking at the documentation right. 
I know the python client library doesn't have this capability (see [this](https://github.com/googleapis/google-cloud-python/issues/9358) issue) but we are authenticating the API differently anyway (using `storage = googleapiclient.discovery.build('storage', 'v1', credentials=credentials)`) . Even in the [issue](https://github.com/lyft/cartography/issues/141) you raised, `project_id` doesn't seem to be present.

Reply:
Good catch!  We didn't include projectnumber in a GCP project at first, but I just submitted https://github.com/lyft/cartography/pull/172 to include it.

After that is merged, you should be able to do a `MERGE` query to connect the project to the storage bucket.

Reply:
Thanks. In the `MERGE` statement, should I create another schema `GCPProjectName`? Since the [current](https://github.com/lyft/cartography/blob/master/docs/schema/gcp.md) one has `id` as `project_id` not `project_number`.